public/guides/SECURITY_GUIDE.md

```diff
@@ -0,0 +1,82 @@
+# GitHub Security Guide
+
+## Repository Access Settings
+For products with sensitive information or proprietary code, set the repository as private to
+maintain confidentiality and control access. For open-source projects or non-sensitive content,
+consider making the repository public to encourage collaboration and transparency.
+
+## Enable Dependency Graph for Private Repositories
```

Comment:
Why only for private repositories?

We could also look to have Dependency Graph enabled by default for every repositories in the SRGSSR organisation: https://docs.github.com/en/code-security/getting-started/securing-your-organization

Reply:
The dependency graph is enabled for public repositories by default (and cannot be disabled AFAIK). Here is an example with our Pillarbox Apple repository:

<img width="1145" alt="Screenshot 2024-03-06 at 09 39 23" src="https://github.com/SRGSSR/guilde-plateformes-propres/assets/170201/c99251b4-bdd4-40e2-92d6-01408519cbfa">